server/src/test/java/io/spine/server/procman/given/ProcessManagerTestEnv.java

```diff
@@ -60,7 +78,129 @@
+        return PmAnswerProblem.newBuilder()
+                              .setExamId(id)
+                              .setAnswer(answer)
+                              .build();
+    }
+
+    public static PmProblemAnswer newProblemAnswer() {
+        return newProblemAnswer(newProblemId(), true);
+    }
+
+    private static PmProblemId newProblemId() {
+        return PmProblemId.newBuilder()
+                          .setId(newUuid())
+                          .build();
+    }
+
+    public static PmProblemAnswer newProblemAnswer(PmProblemId id, boolean correct) {
+        return PmProblemAnswer.newBuilder()
+                              .setProblemId(id)
+                              .setCorrect(correct)
+                              .build();
+    }
+
+    public static TenantId newTenantId() {
+        return TenantId.newBuilder()
+                       .setValue(newUuid())
+                       .build();
+    }
+
+    /**
+     * Creates a new multitenant bounded context with a registered
+     * {@linkplain ExamProcmanRepository exam repository}.
+     */
+    public static BoundedContext newExamBoundedContext() {
+        final BoundedContext boundedContext = BoundedContext.newBuilder()
+                                                            .setMultitenant(true)
+                                                            .build();
+        boundedContext.register(new ExamProcmanRepository());
+        return boundedContext;
+    }
+
+    /**
+     * Creates a new multitenant bounded context with a registered
+     * {@linkplain DirectExamProcmanRepository exam repository}.
+     */
+    public static BoundedContext newDirectExamBoundedContext() {
+        final BoundedContext boundedContext = BoundedContext.newBuilder()
+                                                            .setMultitenant(true)
+                                                            .build();
+        boundedContext.register(new DirectExamProcmanRepository());
+        return boundedContext;
+    }
+
+    /**
+     * A convenience method for closing the bounded context.
+     *
+     * <p>Instead of a checked {@link java.io.IOException IOException}, wraps any issues
+     * that may occur while closing, into an {@link IllegalStateException}.
+     *
+     * @param boundedContext a bounded context to close
+     */
+    public static void closeContext(BoundedContext boundedContext) {
+        checkNotNull(boundedContext);
+        try {
+            boundedContext.close();
+        } catch (Exception e) {
+            throw Exceptions.illegalStateWithCauseOf(e);
+        }
+    }
+
+    private static TestActorRequestFactory requestFactory(TenantId tenantId) {
+        return TestActorRequestFactory.newInstance(AggregateTestEnv.class, tenantId);
+    }
+
+    /**
+     * Reads all events from the bounded context for the provided tenant.
+     */
+    public static List<Event> readAllEvents(final BoundedContext boundedContext,
```

Comment:
Should this methods be moved to `testutils`? 
We‘re using them in `examples-java` already and right now they‘re populating each test environment.

Reply:
@mdrachuk I've browsed `testutil-server` for a good location, but this case is a bit orphaned at the moment and it's difficult to generalize it nicely. 

Let's create `io.spine.server.event.TestEventReader` utility class in `testutil-server`. To my understanding, that's the best and the most unobtrusive way to do it for now.

Reply:
@armiol It‘s not just reading events. It‘s also `newBoundedContext`, `newTenantId`, `closeContext` and `requestFactory`. All of those are identical throughout framework and its example tests.